File: greencab/src/main/java/fr/pantheonsorbonne/ufr27/miage/model/Passenger.java
package fr.pantheonsorbonne.ufr27.miage.model;

import javax.persistence.*;

@Entity
public class Passenger {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "passenger_id", nullable = false)
    private Integer id;

    @Column(name = "first_name", length = 50)
    private String firstName;

    @Column(name = "last_name", length = 50)
    private String lastName;

    @Column(name = "phone", length = 30)
    private String phone;

    @Column(name = "email_address")
    private String emailAddress;

    @Column(name = "password", length = 50)
    private String password;

    @OneToOne(optional = false)
    @JoinColumn(name = "passenger_card", nullable = false)
    private Card passengerCard;

    public Card getPassengerCard() {
        return passengerCard;
    }

    public void setPassengerCard(Card passengerCard) {
        this.passengerCard = passengerCard;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getEmailAddress() {
        return emailAddress;
    }

    public void setEmailAddress(String emailAddress) {
        this.emailAddress = emailAddress;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }
}

File: greencab/src/main/portfolio/src/model/Passenger.ts


export class Passenger {
  private _id: number;
  private _first_name: string;
  private _last_name: string;
  private _phone: string;
  private _emailAddress: string;
  private _password: string;

  constructor(id: number, first_name: string, last_name: string, phone: string, email: string, password: string) {
    this._id = id;
    this._first_name = first_name;
    this._last_name = last_name;
    this._phone = phone;
    this._emailAddress = email;
    this._password = password;
  }

  get id(): number {
    return this._id;
  }

  get first_name(): string {
    return this._first_name;
  }

  get last_name(): string {
    return this.last_name;
  }

  get phone(): string {
    return this._phone;
  }

  get emailAddress(): string {
    return this._emailAddress;
  }

  get password(): string {
    return this._password;
  }
}
